services/supervisor/src/proto/schema.proto

```diff
@@ -18,8 +18,45 @@ package exonum.supervisor;
 
 import "types.proto";
 import "service.proto";
+import "runtime.proto";
+import "google/protobuf/empty.proto";
 
 message ConfigProposalWithHash {
     exonum.crypto.Hash propose_hash = 1;
     ConfigPropose config_propose = 2;
 }
+
+// Information about deploy error.
+message ErrorInfo {
+    // Height on which error occurred.
+    uint64 height = 1;
+    // Occurred error.
+    exonum.runtime.ExecutionError error = 2;
+}
+
+// Reason for a deployment failure.
+message DeployFailCause {
+    oneof type {
+        // Failure caused by deadline reaching.
+        google.protobuf.Empty deadline = 1;
+        // Failure caused by the execution error.
+        ErrorInfo error = 2;
+    }
+}
+
+// Deployment state.
+message DeployState {
+    enum Type {
+        // Deploy is in process.
+        PENDING = 0;
+        // Deployment failed.
+        FAIL = 1;
```

Comment:
What do you think about adding `TIMEOUT` variant for the deadline and getting rid of `DeployFailCause` altogether?